Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -563,6 +563,13 @@ static inline bool objectIsRelayoutBoundary(const RenderElement* object)
     if (object->style().width().isIntrinsicOrAuto() || object->style().height().isIntrinsicOrAuto() || object->style().height().isPercentOrCalculated() || object->style().width().isPercentOrCalculated())
         return false;
 
+    // A block which doesn't establish a new formating context can pass a whole
+    // bunch of state (floats, margins) to an arbitrary sibling, causing that
+    // sibling to position/size differently.
+    auto* renderBlock = dynamicDowncast<RenderBlock>(object);
+    if (!renderBlock->createsNewFormattingContext())
+        return false;
+
```

Comment:
While it makes sense to stop dirty bit propagation at FC boundary (and not within) -and it also matches LFC design, this PR is slightly incomplete as we should also remove certain redundant checks.